File: VB/Dashboard_CreatePivot/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.DashboardCommon

Namespace Dashboard_CreatePivot
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub
		Private Function CreatePivot(ByVal dataSource As IDashboardDataSource) As PivotDashboardItem
			Dim pivot As New PivotDashboardItem()
			pivot.DataSource = dataSource
			pivot.Columns.AddRange(New Dimension("Country"), New Dimension("Sales Person"))
			pivot.Rows.AddRange(New Dimension("CategoryName"), New Dimension("ProductName"))
			pivot.Values.AddRange(New Measure("Extended Price"), New Measure("Quantity"))
			pivot.AutoExpandColumnGroups = False
			pivot.AutoExpandRowGroups = True
			Return pivot
		End Function
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load

			Dim dashboard As New Dashboard()
			Dim dataSource As New DashboardExcelDataSource() With {.FileName = "SalesPerson.xlsx", .SourceOptions = New DevExpress.DataAccess.Excel.ExcelSourceOptions(New DevExpress.DataAccess.Excel.ExcelWorksheetSettings() With {.WorksheetName = "Data", .CellRange = "A1:L100"})}
			dataSource.Fill()
			dashboard.DataSources.Add(dataSource)

			Dim pivot As PivotDashboardItem = CreatePivot(dataSource)
			dashboard.Items.Add(pivot)

			dashboardViewer1.Dashboard = dashboard
			dashboardViewer1.ReloadData()
		End Sub
	End Class
End Namespace

File: CS/Dashboard_CreatePivot/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.DashboardCommon;

namespace Dashboard_CreatePivot {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        private PivotDashboardItem CreatePivot(IDashboardDataSource dataSource) {
            PivotDashboardItem pivot = new PivotDashboardItem();
            pivot.DataSource = dataSource;
            pivot.Columns.AddRange(new Dimension("Country"), new Dimension("Sales Person"));
            pivot.Rows.AddRange(new Dimension("CategoryName"), new Dimension("ProductName"));
            pivot.Values.AddRange(new Measure("Extended Price"), new Measure("Quantity"));
            pivot.AutoExpandColumnGroups = false;
            pivot.AutoExpandRowGroups = true;
            return pivot;
        }
        private void Form1_Load(object sender, EventArgs e) {

            Dashboard dashboard = new Dashboard();
            DashboardExcelDataSource dataSource = new DashboardExcelDataSource()
            {
                FileName = "SalesPerson.xlsx",
                SourceOptions = new DevExpress.DataAccess.Excel.ExcelSourceOptions(
                    new DevExpress.DataAccess.Excel.ExcelWorksheetSettings()
                    {
                        WorksheetName = "Data",
                        CellRange = "A1:L100"
                    }
                    )
            };
            dataSource.Fill();
            dashboard.DataSources.Add(dataSource);

            PivotDashboardItem pivot = CreatePivot(dataSource);
            dashboard.Items.Add(pivot);

            dashboardViewer1.Dashboard = dashboard;
            dashboardViewer1.ReloadData();
        }
    }
}
